### PR TITLE
feat: fscloud profile

### PR DESCRIPTION
**Pull request by shemau**
### Description

Add an FSCloud profile

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

Include an FSCloud profile with KMS encryption and Event Notifications

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Comment by shemau** [comment]
The new FSCloud profile and associated test in schematics passed on this commit a4f2e0f2b57be8f676beb4c9de3d391493acb2ca

File: modules/fscloud/version.tf [hunk 1]
```diff
@@ -0,0 +1,14 @@
+terraform {
+  required_version = ">= 1.3.0, <1.6.0"
```
Comment (ocofaigh) [inline modules/fscloud/version.tf:2]: ">= 1.3.0, <1.7.0"

File: modules/fscloud/main.tf [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+# Get Cloud Account ID
+##############################################################################
+
+data "ibm_iam_account_settings" "iam_account_settings" {
+}
+
+##############################################################################
+# Create CBR Zone
+##############################################################################
+module "cbr_zone" {
```
Comment (ocofaigh) [inline modules/fscloud/main.tf:11]: this module should not create the CBR zone

File: modules/fscloud/main.tf [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  sm_tags                          = var.sm_tags
+  service_endpoints                = "private"
+  endpoint_type                    = "private"
+  kms_encryption_enabled           = true
+  existing_kms_instance_guid       = var.existing_kms_instance_guid
+  enable_event_notification        = true
+  existing_en_instance_crn         = var.existing_en_instance_crn
+  skip_en_iam_authorization_policy = false
+  kms_key_crn                      = var.kms_key_crn
+  cbr_rules = [
```
Comment (ocofaigh) [inline modules/fscloud/main.tf:38]: The module should not create any CBR rules, it should just expose the variable for consumers to configure

File: examples/fscloud/version.tf [hunk 4]
```diff
@@ -0,0 +1,10 @@
+terraform {
+  required_version = ">= 1.3.0, <1.6.0"
```
Comment (ocofaigh) [inline examples/fscloud/version.tf:2]: >= 1.3.0, <1.7.0

File: examples/fscloud/variables.tf [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  type        = bool
+  description = "Whether to use an existing resource group."
+  default     = false
+}
+
+variable "resource_group_name" {
+  type        = string
+  description = "A resource group name to use for this example, if `existing_resource_group` is false a new resource group will be created"
+  default     = "sm-fscloud"
+}
```
Comment (ocofaigh) [inline examples/fscloud/variables.tf:29]: can use be consistent with all other example and just have a variable called `resource_group`:

```
variable "resource_group" {
  type        = string
  description = "The name of an existing resource group to provision resources in to. If not set a new resource group will be created using the prefix variable"
  default     = null
}
```

File: examples/fscloud/variables.tf [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+##############################################################################
+# Event Notification (EN)
+##############################################################################
+
+variable "en_region" {
+  type        = string
+  description = "Region where event notification will be created"
+  default     = "au-syd"
+}
```
Comment (ocofaigh) [inline examples/fscloud/variables.tf:45]: No need to have `en_region` variable - just have the example use same region for all resources

File: modules/fscloud/README.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
+# Profile for IBM Cloud Framework for Financial Services
+
+This code is a version of the [parent root module](../../) that includes a default configuration that complies with the relevant controls from the [IBM Cloud Framework for Financial Services](https://cloud.ibm.com/docs/framework-financial-services?topic=framework-financial-services-about). See the [Example for IBM Cloud Framework for Financial Services](/examples/fscloud/) for logic that uses this module.
+
+The default values in this profile were scanned by [IBM Code Risk Analyzer (CRA)](https://cloud.ibm.com/docs/code-risk-analyzer-cli-plugin?topic=code-risk-analyzer-cli-plugin-cra-cli-plugin#terraform-command) for compliance with the IBM Cloud Framework for Financial Services profile that is specified by the IBM Security and Compliance Center.
+
+The IBM Cloud Framework for Financial Services mandates the application of an inbound network-based allowlist in front of the IBM Cloud Secrets Manager instance. You can comply with this requirement by using the `cbr_rules` variable in the module, which can be used to create a narrow context-based restriction rule that is scoped to the Secrets Manager instance.
+
```
Comment (ocofaigh) [inline modules/fscloud/README.md:8]: Missing the usage section

File: modules/fscloud/main.tf [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+    value = var.vpc_crn,
+  }]
+}
+
+module "secrets_manager" {
+  source                           = "../.."
+  resource_group_id                = var.resource_group_id
+  region                           = var.region
+  secrets_manager_name             = var.secrets_manager_name #tfsec:ignore:general-secrets-no-plaintext-exposure
+  sm_service_plan                  = "standard"
```
Comment (ocofaigh) [inline modules/fscloud/main.tf:28]: this needs to be a variable, and ensure in our tests we set it to trial (the default value of the module can be standard though)

File: modules/fscloud/main.tf [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+  sm_tags                          = var.sm_tags
+  service_endpoints                = "private"
+  endpoint_type                    = "private"
+  kms_encryption_enabled           = true
+  existing_kms_instance_guid       = var.existing_kms_instance_guid
+  enable_event_notification        = true
```
Comment (ocofaigh) [inline modules/fscloud/main.tf:34]: I don't think having event notifications is an fscloud requirement? If not, expose as a variable and make it optional maybe

File: modules/fscloud/main.tf [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+  existing_en_instance_crn         = var.existing_en_instance_crn
+  skip_en_iam_authorization_policy = false
```
Comment (ocofaigh) [inline modules/fscloud/main.tf:36]: this should be exposed as a variable

File: modules/fscloud/outputs.tf [hunk 11]
```diff
@@ -0,0 +1,4 @@
+output "secrets_manager_guid" {
+  value       = module.secrets_manager.secrets_manager_guid
+  description = "GUID of Secrets-Manager instance in which IAM engine was configured"
+}
```
Comment (ocofaigh) [inline modules/fscloud/outputs.tf:4]: please add more ouputs (id, name, crn etc etc)

**Review by ocofaigh — CHANGES_REQUESTED** [review]
@shemau see my comments

File: tests/pr_test.go [hunk 12]
```diff
@@ -74,37 +74,39 @@ func TestRunUpgradeExample(t *testing.T) {
...
+		Region:                 region,
 	})
 
-	return options
-}
-
-func TestRunPrivateExample(t *testing.T) {
-	t.Parallel()
+	// Workaround for https://github.com/IBM-Cloud/terraform-provider-ibm/issues/5154
+	options.AddWorkspaceEnvVar("IBMCLOUD_KP_API_ENDPOINT", "https://private."+region+".kms.cloud.ibm.com", false, false)
```
[the remark this answers is not in the document]
Comment (shemau) [reply tests/pr_test.go]: Removed

File: modules/fscloud/version.tf (hunk 1 above)
Comment (shemau) [reply modules/fscloud/version.tf]: Updated

File: modules/fscloud/main.tf (hunk 2 above)
Comment (shemau) [reply modules/fscloud/main.tf]: Relocated CBR to example

File: modules/fscloud/main.tf (hunk 3 above)
Comment (shemau) [reply modules/fscloud/main.tf]: Relocated CBR to example

File: examples/fscloud/version.tf (hunk 4 above)
Comment (shemau) [reply examples/fscloud/version.tf]: Updated

File: modules/fscloud/README.md (hunk 7 above)
Comment (shemau) [reply modules/fscloud/README.md:8]: Included

File: modules/fscloud/main.tf (hunk 9 above)
Comment (shemau) [reply modules/fscloud/main.tf]: variable

File: modules/fscloud/main.tf (hunk 10 above)
Comment (shemau) [reply modules/fscloud/main.tf]: variable

File: modules/fscloud/outputs.tf (hunk 11 above)
Comment (shemau) [reply modules/fscloud/outputs.tf:23]: Adopted same outputs as root module

File: modules/fscloud/main.tf (hunk 8 above)
Comment (shemau) [reply modules/fscloud/main.tf]: variable added

File: examples/fscloud/variables.tf (hunk 5 above)
Comment (shemau) [reply examples/fscloud/variables.tf:29]: removed name

File: examples/fscloud/variables.tf (hunk 6 above)
Comment (shemau) [reply examples/fscloud/variables.tf]: removed

**Pull request merged**